### DISCUSSION TITLE
关于1p易用性的一些建议

### DISCUSSION THREAD

**Discussion by defeatedperson** [1Panel-dev/1Panel#4161] [Ideas]
### 1Panel 版本

1.10

### 请描述您的需求或者改进建议

1.
小白入门1p的三座大山(括号内容为小白第一印象)
例如:
php站点目录权限(为啥无法上传文件？不清楚需要设置)，
数据库地址(为啥连不上数据库？不是local host或者127.0.0.1吗？)，
php端口(这是啥？没听过，需要改吗？)

2.
1p的ssh指令，确实不友好，我最开始都看的比较懵，不知道怎么输入

### 请描述你建议的实现方案

1.在部分页面增加跳转1p论坛按钮(数据库页面/站点概览页面/网站创建页面，等)
现在bt貌似加了不少这种按钮，并且安装失败的提示也改成:
xx安装失败，请联系QQ备注xx问题，会尽快处理。

感觉可以降低操作上面的问题，以及对1p产生误会。

2.
推荐改成:

输入:1p
输出:
1.查看默认1panel信息
2.修改面板用户名
3.修改面板密码
4.关闭mfa
5.重启面板
6.关闭面板ssh
7.修复面板

用户这个时候，只需要输入数字即可。

也可以一开始就输入1p1
自动执行1命令

### 附加信息

这些对稍微折腾一点的用户，基本没啥问题
但对于刚入门的小白，或者从bt迁移过来的朋友，
有这些提示会方便不少，也能留下不错第一印象，可以得到推荐

**Comment by realzolo** [comment]
> ### 1Panel 版本
> 1.10
> 
> ### 请描述您的需求或者改进建议
> 
> 小白入门1p的三座大山(括号内容为小白第一印象) 例如: php站点目录权限(为啥无法上传文件？不清楚需要设置)， 数据库地址(为啥连不上数据库？不是local host或者127.0.0.1吗？)， php端口(这是啥？没听过，需要改吗？)
> 
> 
> 1p的ssh指令，确实不友好，我最开始都看的比较懵，不知道怎么输入
> 
> ### 请描述你建议的实现方案
> 1.在部分页面增加跳转1p论坛按钮(数据库页面/站点概览页面/网站创建页面，等) 现在bt貌似加了不少这种按钮，并且安装失败的提示也改成: xx安装失败，请联系QQ备注xx问题，会尽快处理。
> 
> 感觉可以降低操作上面的问题，以及对1p产生误会。
> 
> 
> 推荐改成:
> 
> 输入:1p 输出: 1.查看默认1panel信息 2.修改面板用户名 3.修改面板密码 4.关闭mfa 5.重启面板 6.关闭面板ssh 7.修复面板
> 
> 用户这个时候，只需要输入数字即可。
> 
> 也可以一开始就输入1p1 自动执行1命令
> 
> ### 附加信息
> 这些对稍微折腾一点的用户，基本没啥问题 但对于刚入门的小白，或者从bt迁移过来的朋友， 有这些提示会方便不少，也能留下不错第一印象，可以得到推荐

对于你提的建议，我作为一个用户来说，是强烈反对的。
1.  数据库页面/站点概览页面/网站创建页面等增加跳转论坛这种与功能无关的按钮，提示信息中包含QQ等联系方式是一个巨大的垃圾设计，对于强迫症来说真的是灾难。对于一个完美的项目来说，保持功能的纯洁性比什么都重要。
2. 直接联系QQ等方式不仅对开发人员造成了极大的负担，且相同问题处理记录无法留痕供其他用户参考。有问题提ISSUE即可。
3. 对于现在的命令行指令，我个人认为是非常符合设计规范的。

我选择1pannel的原因就是因为1p的设计比较现代化，功能纯净，不像bt广告满天飞，除了功能多，设计真的不咋地。
以上是我自己的理解。

**Comment by defeatedperson** [comment]
> > ### 1Panel 版本
> > 1.10
> > ### 请描述您的需求或者改进建议
> > 小白入门1p的三座大山(括号内容为小白第一印象) 例如: php站点目录权限(为啥无法上传文件？不清楚需要设置)， 数据库地址(为啥连不上数据库？不是local host或者127.0.0.1吗？)， php端口(这是啥？没听过，需要改吗？)
> > 1p的ssh指令，确实不友好，我最开始都看的比较懵，不知道怎么输入
> > ### 请描述你建议的实现方案
> > 1.在部分页面增加跳转1p论坛按钮(数据库页面/站点概览页面/网站创建页面，等) 现在bt貌似加了不少这种按钮，并且安装失败的提示也改成: xx安装失败，请联系QQ备注xx问题，会尽快处理。
> > 感觉可以降低操作上面的问题，以及对1p产生误会。
> > 推荐改成:
> > 输入:1p 输出: 1.查看默认1panel信息 2.修改面板用户名 3.修改面板密码 4.关闭mfa 5.重启面板 6.关闭面板ssh 7.修复面板
> > 用户这个时候，只需要输入数字即可。
> > 也可以一开始就输入1p1 自动执行1命令
> > ### 附加信息
> > 这些对稍微折腾一点的用户，基本没啥问题 但对于刚入门的小白，或者从bt迁移过来的朋友， 有这些提示会方便不少，也能留下不错第一印象，可以得到推荐
> 
> 对于你提的建议，我作为一个用户来说，是强烈反对的。
> 
> 1. 数据库页面/站点概览页面/网站创建页面等增加跳转论坛这种与功能无关的按钮，提示信息中包含QQ等联系方式是一个巨大的垃圾设计，对于强迫症来说真的是灾难。对于一个完美的项目来说，保持功能的纯洁性比什么都重要。
> 2. 直接联系QQ等方式不仅对开发人员造成了极大的负担，且相同问题处理记录无法留痕供其他用户参考。有问题提ISSUE即可。
> 3. 对于现在的命令行指令，我个人认为是非常符合设计规范的。
> 
> 我选择1pannel的原因就是因为1p的设计比较现代化，功能纯净，不像bt广告满天飞，除了功能多，设计真的不咋地。 以上是我自己的理解。

您貌似误会了，
我的意思是，
1.在这些地方增加:跳转1p论坛的按钮，
而不是开发者QQ联系方式，
因为部分用户不知道去哪里查看这些，例如一个？形状圆形按钮，几乎不会影响美观，用户也能知道不懂点这个

2.对于ssh命令，至少我了解的很多用户(少说几十位)，压根不知道怎么用，并不是每个人都很懂

3.我的渠道比较丰富，毕竟我也是一个服务器代理+视频博主+自己折腾网站，可以了解的多一点(在普通用户当中)

感谢你的回应，但请您冷静思考，好的地方要学习，差的地方要弥补，不能一味排外

**Comment by ChishFoxcat** [comment]
1. 1panel的php运行环境是按照docker方式运行，则需要端口映射方式
2. 数据库自动跳转的地址应为手动设置后的地址
3. ssh终端确实不怎么好用，可以使用其他软件代替
4. 纯净的面板不应是什么外部链接都一股脑塞进去，此仓库已经提供了很完善的提问系统

> 不要说什么小白那个小白，建议多去学习和适应，不是所有东西都能满足所有人的习惯

**Comment by defeatedperson** [comment]
> 1. 1panel的php运行环境是按照docker方式运行，则需要端口映射方式
> 2. 数据库自动跳转的地址应为手动设置后的地址
> 3. ssh终端确实不怎么好用，可以使用其他软件代替
> 4. 纯净的面板不应是什么外部链接都一股脑塞进去，此仓库已经提供了很完善的提问系统
> 
> > 不要说什么小白那个小白，建议多去学习和适应，不是所有东西都能满足所有人的习惯



> 1. 1panel的php运行环境是按照docker方式运行，则需要端口映射方式
> 2. 数据库自动跳转的地址应为手动设置后的地址
> 3. ssh终端确实不怎么好用，可以使用其他软件代替
> 4. 纯净的面板不应是什么外部链接都一股脑塞进去，此仓库已经提供了很完善的提问系统
> 
> > 不要说什么小白那个小白，建议多去学习和适应，不是所有东西都能满足所有人的习惯

链接确实不能乱塞，但适当的提示，总是要加一点，
请您冷静思考一下，遇到一个新产品，没有任何提示，您会知道去那里查资料吗？
部分输入位置，增加？的圆形小按钮提示用户，
这样基本不影响美观，又能提升用户好感度，让用户更放心去宣传
(毕竟宣传给他人，你基本就需要对其进行解答疑问，有了适当的提示，推荐者才能放心推荐，不必担心上手困难而累死自己)

当然，我也希望1p有更清爽的外观，但这与易用性并不冲突，在用户反应较多的地方增加适应的提示，利大于弊

以上是我的理解，感谢您参与讨论

**Reply by ChishFoxcat** [reply]
明白，这些都是为了大部分人更好的使用

1. 底下有文档的链接
2. 提示用户感觉这是之后的事了，毕竟也不是给新手用户用的，只是方便运营者好运行他自己的服务器罢了
3. 便捷直白明了，也确实不清

有发声说明他的确有地方不足，但他们确实还在改，这你不必太过担心，我也是从一开始就用的
